### PR TITLE
fixed #2263, geth keys with ciphertext shorter than 32 bytes

### LINKED ISSUE #2263: Geth keys with ciphertext of 31 bytes (62 hex digits) length
Originally reported on slack

Not only they fail to load individually, such keys also prevent loading of any key in the same directory (`listAccount` fails)

It's also unclear how the geth produces and reads this kind of keys

User with large amount of keys reports that around 0.37% of all keys have 62 hex digits (252 bits) in the `ciphertext` section of the key file

### PR DESCRIPTION
**Pull request by debris**
Fixes https://github.com/ethcore/parity/issues/2263

We also need to add additional tests. @NikVolf can you send me keyfile with short ciphertext?

### CODE REVIEW

File: ethstore/src/json/bytes.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+use std::{ops, str};
+use serde::{Deserialize, Deserializer, Error, Serialize, Serializer};
+use rustc_serialize::hex::{ToHex, FromHex, FromHexError};
+
+#[derive(Debug, PartialEq)]
+pub struct Bytes(Vec<u8>);
```
Comment (rphmeier) [inline ethstore/src/json/bytes.rs]: this is the third definition of this type afaict (`rpc`, `json`, and now here) -- is it possible to be more DRY?

File: ethstore/src/json/hash.rs [hunk 2]
```diff
@@ -92,6 +90,12 @@ macro_rules! impl_hash {
 			}
 		}
 
+		impl From<&'static str> for $name {
+			fn from(s: &'static str) -> Self {
+				s.parse().unwrap()
```
Comment (rphmeier) [inline ethstore/src/json/hash.rs]: `.expect("invalid string literal for {}: '{}'", stringify!($name), s)` would be a little more helpful

File: ethstore/src/json/bytes.rs (hunk 1 above)
Comment (debris) [inline ethstore/src/json/bytes.rs]: They share the same name, but they are not the same. `json::Bytes` deserialization is extremely lenient, `rpc::Bytes` deserialization allows `0x` prefixes. Here, the format is very strict

File: ethstore/src/json/hash.rs (hunk 2 above)
Comment (debris) [inline ethstore/src/json/hash.rs]: 👍

**Comment by gavofyork** [comment]
CI failing.

**Comment by debris** [comment]
CI failure is unrelated, I restarted it

**Comment by gavofyork** [comment]
would be nice to have a test; should be merged for 1.3.2 regardless, but hopefully @NikVolf can get a test in before.

**Comment by NikVolf** [comment]
according to Gustav, test case can be obtained by generating enough keys in geth
just keep creating until you encounter short one or use Gustav script there
https://github.com/ethcore/parity/issues/2263

`godep go test -run TestKeyStorePassphrase -count 500`

**Comment by debris** [comment]
I tried that script, but keys are not printed to the output by default. I asked Gustav for help with that. I will update the pr as soon as he responds :)

**Comment by Gustav-Simonsson** [comment]
Added two test vectors for keys with length  31 and 30: https://github.com/ethereum/go-ethereum/pull/3032/files

**Comment by debris** [comment]
tests are failing, I'm working on fix now :)

**Pull request merged**